api/src/test/java/marquez/DatasetIntegrationTest.java

```diff
@@ -408,36 +383,118 @@ public void testApp_showsDeletedDatasetAfterReceivingNewVersion() throws IOExcep
-            .whenComplete(
-                (val, error) -> {
-                  if (error != null) {
-                    Assertions.fail("Could not complete request");
-                  }
-                });
-
-    // Ensure the event was correctly rejected and a proper response code returned.
+    CompletableFuture<Integer> resp = sendEvent(event);
     assertThat(resp.join()).isEqualTo(201);
 
     client.deleteDataset(namespace, name);
 
     List<Dataset> datasets = client.listDatasets(namespace);
     assertThat(datasets).hasSize(0);
-    resp =
-        this.sendLineage(Utils.toJson(event))
-            .thenApply(HttpResponse::statusCode)
-            .whenComplete(
-                (val, error) -> {
-                  if (error != null) {
-                    Assertions.fail("Could not complete request");
-                  }
-                });
-
+    resp = sendEvent(event);
     assertThat(resp.join()).isEqualTo(201);
 
     datasets = client.listDatasets(namespace);
     assertThat(datasets).hasSize(1);
   }
+
+  @Test
+  public void testApp_doesNotShowDeletedDatasetAfterDeleteNamespace() throws IOException {
+    String namespace = "namespace";
+    String name = "table";
+    LineageEvent event =
+        new LineageEvent(
+            "COMPLETE",
+            Instant.now().atZone(ZoneId.systemDefault()),
+            new LineageEvent.Run(UUID.randomUUID().toString(), null),
+            new LineageEvent.Job("namespace", "job_name", null),
+            List.of(new LineageEvent.Dataset(namespace, name, LineageTestUtils.newDatasetFacet())),
+            Collections.emptyList(),
+            "the_producer");
+
+    final CompletableFuture<Integer> resp = sendEvent(event);
+    assertThat(resp.join()).isEqualTo(201);
+
+    client.deleteNamespace(namespace);
+
+    List<Dataset> datasets = client.listDatasets(namespace);
+    assertThat(datasets).hasSize(0);
+  }
+
+  @Test
+  public void testApp_doesNotShowDeletedDatasetAfterUndeleteNamespace() throws IOException {
+    String namespaceName = "namespace";
+    String name = "table";
+
+    LineageEvent firstEvent =
+        new LineageEvent(
+            "COMPLETE",
+            Instant.now().atZone(ZoneId.systemDefault()),
+            new LineageEvent.Run(UUID.randomUUID().toString(), null),
+            new LineageEvent.Job(namespaceName, "job_name", null),
+            List.of(
+                new LineageEvent.Dataset(namespaceName, name, LineageTestUtils.newDatasetFacet())),
+            Collections.emptyList(),
+            "the_producer");
+
+    LineageEvent secondEvent =
+        new LineageEvent(
+            "COMPLETE",
+            Instant.now().atZone(ZoneId.systemDefault()),
+            new LineageEvent.Run(UUID.randomUUID().toString(), null),
+            new LineageEvent.Job(namespaceName, "second_job_name", null),
+            List.of(
+                new LineageEvent.Dataset(namespaceName, name, LineageTestUtils.newDatasetFacet())),
```

Comment:
Isn't it the same dataset as before?
If so, test cannot return two datasets regardless of delete namespace behavior.